File: core/cli-help.js
/**
 * Shared CLI helpers for skill API scripts.
 */

function wantsHelp(argv) {
  return argv.includes('--help') || argv.includes('-h');
}

function printHelp(scriptName, description, commands) {
  console.log(`${scriptName}`);
  if (description) console.log(`${description}\n`);
  console.log('Usage:');
  for (const cmd of commands) {
    const aliases = cmd.aliases ? ` (aliases: ${cmd.aliases.join(', ')})` : '';
    console.log(`  ${cmd.usage}${aliases}`);
    if (cmd.detail) console.log(`    ${cmd.detail}`);
  }
  console.log('\nOptions:');
  console.log('  --help, -h    Show this help');
}

module.exports = { wantsHelp, printHelp };

File: core/cli_help.py
"""Shared CLI helpers for skill API scripts."""


def wants_help(argv):
    return "--help" in argv or "-h" in argv


def print_help(script_name, description, commands):
    print(script_name)
    if description:
        print(f"{description}\n")
    print("Usage:")
    for cmd in commands:
        aliases = f" (aliases: {', '.join(cmd['aliases'])})" if cmd.get("aliases") else ""
        print(f"  {cmd['usage']}{aliases}")
        if cmd.get("detail"):
            print(f"    {cmd['detail']}")
    print("\nOptions:")
    print("  --help, -h    Show this help")